### PR TITLE
[Implement] Buffer.swap16/32/64

### PR DESCRIPTION
**Pull request by RedDwarfian**

### CODE REVIEW

**Comment by jtenner** [comment]
@dcodeIO?

**Comment by jtenner** [comment]
LGTM.

File: assembly/buffer/index.ts [hunk 1]
```diff
@@ -237,6 +237,45 @@ export class Buffer extends Uint8Array {
...
     return offset + 8;
   }
+
+  swap16(): Buffer {
+    let dataLength = this.dataLength;
+    // Make sure dataLength is even
+    if (dataLength & 1) throw new RangeError(E_INVALIDLENGTH);
+    let dataStart = this.dataStart;
+    dataLength += dataStart;
+    while (dataStart < dataLength) {
```
Comment (MaxGraey) [inline assembly/buffer/index.ts]: I guess better use `while (dataStart < dataLength + dataStart)` instead `dataLength += dataStart`

**Comment by MaxGraey** [comment]
LGTM

**Review by jtenner — APPROVED** [review]

**Pull request merged**